src/nvim/popupmenu.c

```diff
@@ -1319,6 +1320,14 @@ static void pum_position_at_mouse(int min_width)
   int col = mouse_col;
   pum_win_row_offset = 0;
   pum_win_col_offset = 0;
+
+  if (ui_has(kUIMultigrid) && (grid == 0 || grid == 1)) {
+    win_T *wp = mouse_find_win(&grid, &row, &col);
+    // mouse_find_win compensates for the offset due to winbar for example
+    // but we don't want that here
+    row += wp->w_winrow_off;
+    col += wp->w_wincol_off;
```

Comment:
Should these lines be added to `pum_select_mouse_pos()` as well?

Reply:
No, I don't think so, in `pum_select_mouse_pos`, the result is used to calculate grid relative coordinates, while here the row and col are absolute ones.

I also tested with this to make sure, and tests started to fail
```c
if (grid == 0) {
  win_T *wp = mouse_find_win(&grid, &row, &col);
  if (ui_has(kUIMultigrid)) {
    // mouse_find_win compensates for the offset due to winbar for example
    // but we don't want that here
    row += wp->w_winrow_off;
    col += wp->w_wincol_off;
  }
}
```

Reply:
Hmm wait. See below.